### ISSUE TITLE
store a shuffle in a variable?

### ISSUE THREAD

**Issue by darkpony** [inkle/ink#301]
is this a thing? would this ever be a doable thing?

is it possible to store the result of a shuffle in a variable?
var randomName = {~inky|blinky|clyde}

**Comment by joningold** [comment]
So, in general, you can achieve this kind of thing by using return values
from functions and storing those.. but as it happens, this example works
fine because strings in ink are actually *ink* not strings. So

~ temp randomName ="{~inky|blinky|clyde}"

...does what you want it to.

cheers
jon

**Comment by facelesspanda** [comment]
wait String variables are treated as content ??
so
```ink
VAR x = " + { cond } global[] sticky choice"
  + local[] sticky choice
  {x}
```
would be legal and functional ? 🤔

**Comment by joethephish** [comment]
Nope, only a subset of ink works within the strings - what we call *inline logic*. So basically, it's all the logic between `{ ... }` that goes on the same line, but not choices, any other flow structuring, or multiline logic. So off the top of my head, that's inline conditions, sequences and printing of values.

Somewhat crazily, you can nest strings and inline logic as much as you like. i.e. the following are equivalent:

`VAR x = "five"`

1.  `"I have {x} apples"`
2. `"I have {"five"} apples"` (The inner quotes are nested *within* the outer quotes, like parentheses.)
3. `" I have {"{"five"}"} apples"`
4. ...and so on!

**Comment by joethephish** [comment]
Oh, but you can call functions from within strings, so if you want to have multiline logic (but not flow structuring since it's disallowed in functions), you can do:

```
VAR num = 3

{"{printNum(num)}"}     // prints three

== function printNum(n) ==
{ n:
  - 1: ~ return "one"
  - 2: ~ return "two"
  - 3: ~ return "three"
  - 4: ~ return "four"
}
```

**Comment by facelesspanda** [comment]
Interesting!
Weird, and my programming instincts scream in pain, but interesting!

**Comment by joethephish** [comment]
Think of it as powerful string interpolation!
